### ISSUE TITLE
[FEATURE] add databricks_metastore data source to retrieve an existing metastore

### ISSUE THREAD

**Issue by nicolasbailly1** [databricks/terraform-provider-databricks#1651]
### Use-cases
Hello,

I need to retrieve an existing metastore to attach it to a new workspace.
Currently, we only can create a new metatstore.

### Proposal

```terraform
data "databricks_metastore" "this" {
  name = "primary"
}

output "metastore_id" {
  value = data.databricks_metastore.this.id
}
```

**Comment by guillesd** [comment]
Hi there! This is much needed for my client... If you want to share metastores among workspace deployments I don't wanna be hardcoding metastore id values!

@nkvuong why do you label this issue as `needs-api`? I checked the Databricks CLI implementation and there's seems to be an endpoint to list metastores https://github.com/databricks/databricks-cli/blob/701959037eac5fa6f168f47af546545609db116e/databricks_cli/unity_catalog/uc_service.py#L44

**Comment by nkvuong** [comment]
@guillesd that endpoint in the CLI implementation is a workspace-specific one, what we need is an account-level endpoint (it should be available some time this month is what we last heard from eng)

**Comment by guillesd** [comment]
@nkvuong but even the provider being used right now to create the metastore is at the workspace level, so I don't see any problem on extending this with the functionality to fetch a metastore as a data source. I'm aware that the idea is to migrate the metastore implementation to the account level APIs but in the meantime this could be quite a quick win in my view

**Comment by alyssons-db** [comment]
This has become an issue for one of my customers as well. It would be great to have a data source for Metastores even if it means using the workspace APIs as @guillesd mentioned

**Comment by waskelton4** [comment]
This feature would really help us out with our TF deployments as well!

**Comment by scheleaap** [comment]
This is an issue for us as well. It would be very helpful to be able to retrieve the metastore instead of having to hard code the ID.

**Comment by FantomKun** [comment]
+1 on behalf of my customer, we have to use http to retrieve the metastore id which is very heavy to operate.

**Comment by nkvuong** [comment]
Closes by #2492
